eng/code-quality-reports/src/test/resources/com/azure/tools/checkstyle/checks/GoodLoggerCheck/WrongClassInLoggerConstructor.java

```diff
@@ -0,0 +1,6 @@
+import com.azure.core.util.logging.ClientLogger;
+
+public class WrongClassInLoggerConstructor {
+    // wrong class in ClientLogger constructor
+    private final ClientLogger logger = new ClientLogger(XXXXXX.class);
```

Comment:
Does this class actually compile - XXXXX.class presumably doesn't exist?

Reply:
These java files in the resource folder don't need to be compiled. It is treated as a text file. CheckStyle just parse the text to be Abstract Syntax Tree and valid with the rules.